### PR TITLE
Thread --trace-parser option through parser

### PR DESCRIPTION
**Pull request by jez**
### Motivation

This will make it easier to visualize what's happening inside the parser and
debug it.

### Test plan

See included automated tests.

### CODE REVIEW

File: third_party/parser/cc/capi.cc [hunk 1]
```diff
@@ -1,101 +1,69 @@
-#include <ruby_parser/capi.hh>
```
Comment (jez) [inline third_party/parser/cc/capi.cc:1]: i found a couple of files that were not having clang-format run on them

i landed those changes as a separate commit, if you want to ignore this noise

i haven't looked into why they're not getting formatted automatically (but it's probably not that important)

File: test/cli/trace_parser/trace_parser.out [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+-> $$ = nterm program ()
+Stack now 0
+Entering state 71
+Next token is token "end of file" ()
+Shifting token "end of file" ()
+Entering state 339
+Cleanup: popping token "end of file" ()
+Cleanup: popping nterm program ()
+test/cli/trace_parser.rb: File Not Found https://srb.help/1004
+Errors: 1
```
Comment (aprocter-stripe) [inline test/cli/trace_parser/trace_parser.out:30]: Is the File Not Found expected here?

**Review by aprocter-stripe — APPROVED** [review]
LGTM with one question about the test output

File: test/cli/trace_parser/trace_parser.out (hunk 2 above)
Comment (jez) [reply test/cli/trace_parser/trace_parser.out:30]: no it's not thanks

Comment (jez) [reply test/cli/trace_parser/trace_parser.out]: (fixed)
